### PR TITLE
feat: Introduced section-based accordion groups to the App Settings interface 

### PR DESCRIPTION
**Pull request by Dnouv**
## Proposed changes (including videos or screenshots)

- Added accordion groups to organize settings into sections, using the `useAppTranslation` hook for section titles.
- Created a new `useAppTranslation` hook to handle app-specific translations, improving code reuse and maintainability.
- Removed the local `useAppTranslation` function and replaced it with the new `useAppTranslation` hook.

After changes:

## Further comments

- Please note that App Developers can customize the `i18n` shown on the Accordion title by defining a key under the Apps `i18n.json`; for example if I have `section: model` (in App settings) defining the following key
```
{
...
"model": "Model Section"
...
}
```
Will render the Accordion with the title as "Model Section" if no key is defined, it will show up as `model`
- [RCAI3-14](https://rocketchat.atlassian.net/browse/RCAI3-14)

[RCAI3-14]: https://rocketchat.atlassian.net/browse/RCAI3-14?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppSettings/AppSettings.tsx [hunk 1]
```diff
@@ -1,26 +1,46 @@
...
+import React, { useMemo } from 'react';
 
 import AppSetting from './AppSetting';
 import type { ISettings } from '../../../../../apps/@types/IOrchestrator';
+import { useAppTranslation } from '../../../hooks/useAppTranslation';
 
 const AppSettings = ({ settings }: { settings: ISettings }) => {
-	const { t } = useTranslation();
+	const appId = useRouteParameter('id');
+	console.log(appId);
```
Comment (tassoevan) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppSettings/AppSettings.tsx:11]:
```suggestion
```

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppSettings/AppSettings.tsx [hunk 2]
```diff
@@ -1,26 +1,46 @@
...
-					{Object.values(settings).map((field) => (
-						<AppSetting key={field.id} {...field} />
-					))}
-				</FieldGroup>
-			</Box>
-		</>
+		<Box display='flex' flexDirection='column' maxWidth='x640' w='full' marginInline='auto'>
+			<Accordion>
+				{groupedSettings.map(([section, sectionSettings]) => (
+					<Accordion.Item key={section} title={tApp(section)} defaultExpanded>
```
Comment (tassoevan) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppSettings/AppSettings.tsx:34]: Shouldn't it be only the first one to be `defaultExpanded`?

Comment (tassoevan) [inline apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppSettings/AppSettings.tsx:34]: Prefer `AccordionItem` over `Accordion.Item`. We're deprecating namespaced components in Fuselage.

**Review by tassoevan — CHANGES_REQUESTED** [review]

File: apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppSettings/AppSettings.tsx (hunk 2 above)
Comment (Dnouv) [reply apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppSettings/AppSettings.tsx:33]: Oh, right, that's the default behavior in the setting; I will fix this.

Comment (Dnouv) [reply apps/meteor/client/views/marketplace/AppDetailsPage/tabs/AppSettings/AppSettings.tsx]: Thank you, resolved!

**Review by tassoevan — APPROVED** [review]

**Pull request merged**